File: python/solutions/binary_search.py
from environment import Solver

class BinarySearchSolver(Solver):
    def solve(self, **input_data):
        """
        Input expected: lista=list sorted ascendent, target=value to search
        Output: index of value to search in list or -1
        """
        lista = input_data.pop("lista")
        target = input_data.pop("target")

        low = 0
        high = len(lista) - 1

        while low <= high:
            mid = (low + high) // 2
            guess = lista[mid]

            if guess == target:
                return mid  # index of position

            if guess > target:
                high = mid-1
            else:
                low = mid+1

        return -1

File: go/solutions/binary_search.go
package solutions

type BinarySearchSolver struct{}

func (s BinarySearchSolver) Solve(input map[string]any) any {
	// arr = array, target = target to find
	arr := input["arr"].([]int)  // type assertion
	target := input["target"].(int)

	var high int = len(arr) - 1
	var low int = 0

	var guess, mid int

	for low <= high {
		mid = (low + high) / 2
		guess = arr[mid]

		if guess == target {
			return mid
		}

		if guess < target {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}

	return -1
}
